rc/lsp.kak

```diff
@@ -228,18 +228,157 @@ define-command lsp-hover -docstring "Request hover info for the main cursor posi
     lsp-did-change-and-then lsp-hover-request
 }
 
-define-command -hidden lsp-hover-request -docstring "Request hover info for the main cursor position" %{
+define-command -hidden lsp-hover-request -params 0..2 -docstring "Request hover info for the main cursor position" %{
     nop %sh{ (printf '
 session   = "%s"
 client    = "%s"
 buffile   = "%s"
 filetype  = "%s"
 version   = %d
 method    = "textDocument/hover"
-[params.position]
-line      = %d
-column    = %d
-' "${kak_session}" "${kak_client}" "${kak_buffile}" "${kak_opt_filetype}" "${kak_timestamp}" ${kak_cursor_line} ${kak_cursor_column} | eval ${kak_opt_lsp_cmd} --request) > /dev/null 2>&1 < /dev/null & }
+[params]
+position.line = %d
+position.column = %d
+style = "%s"
+context = "%s"
+' "${kak_session}" "${kak_client}" "${kak_buffile}" "${kak_opt_filetype}" "${kak_timestamp}" ${kak_cursor_line} ${kak_cursor_column} "$1" "$2" | eval ${kak_opt_lsp_cmd} --request) > /dev/null 2>&1 < /dev/null & }
+}
+
+declare-option -hidden str lsp_symbol_kind_completion %{
+  symbol_kinds="
+  File Module Namespace Package Class Method Property Field Constructor Enum Interface Function
+  Variable Constant String Number Boolean Array Object Key Null EnumMember Struct Event Operator
+  TypeParameter
+  "
+  for symbol_kind in ${symbol_kinds}; do
+    printf '%s\n' "${symbol_kind}"
```

Comment:
The loop is actually not necessary - `printf '%s\n' ${symbol_kinds}` is enough

Reply:
I tried doing it the way you suggested. Each completion option needs to be separated by a newline so the `for` loop does seem necessary.

We probably can avoid the `for` via some smart bash-ism though.

Reply:
You wrote `printf '%s\n' "${symbol_kinds}"` but it should be `printf '%s\n' ${symbol_kinds}`.
(We need to word-split somewhere, whether in the for-loop or in the args to printf. I just find the implicit looping quite elegant)

Reply:
Nice!